### PR TITLE
DEVX-1274-yb: make two distinct workflows for connectors

### PR DESCRIPTION
**Pull request by ybyzek**

### CODE REVIEW

File: kubernetes/docs/includes/deploy-jars-k8s.rst [hunk 1]
```diff
@@ -1,36 +1,23 @@
-Deploying Custom Kafka Connect JARs with Operator 
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Deploying Connectors with Operator 
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-`Kafka Connect <https://docs.confluent.io/current/connect/index.html>`__ employs an extensible programming model to faciliate `Data Conversions <https://docs.confluent.io/current/connect/concepts.html#converters>`__, `Single Message Transformations (SMTs) <https://docs.confluent.io/current/connect/concepts.html#transforms>`__, or the development of custom connectors if a `supported connector <https://www.confluent.io/hub/>`__ is not available.  The Connect `packaging model <https://docs.confluent.io/current/connect/devguide.html#packaging>`__ will require that you deploy an archive containing your custom JAR to the `Kafka Connect's plugin path <https://docs.confluent.io/current/connect/userguide.html#connect-installing-plugins>`__.
+|c-hub| is an online library of pre-packaged and ready-to-install connectors, transformations, and converters that suit your needs.
```
Comment (rspurgeon) [inline kubernetes/docs/includes/deploy-jars-k8s.rst]: what kinda stinks about the doc includes is there is no link... i think a link to CHub would be helpful here

File: kubernetes/docs/includes/deploy-jars-k8s.rst [hunk 2]
```diff
@@ -1,36 +1,23 @@
...
+Once you build the custom Docker image, Kubernetes will need to pull this image from a Docker Registry to create the Pods.
 
-This demonstration utilizes a `custom Connector to generate mock events <https://github.com/confluentinc/kafka-connect-datagen>`__ which is **not** installed by default in the Confluent Kafka Connect Docker images.  In order to utilize this plugin, we build a custom Docker image which is derived from the Confluent Operator based Docker image and install the custom connector using the ``confluent-hub`` client.
+For more advanced use cases where you want to use a custom connector instead of a pre-packaged one available at |c-hub|, you may create a Docker image with a custom connector from a `local archive <https://docs.confluent.io/current/connect/managing/confluent-hub/command-reference/confluent-hub-install.html#confluent-hub-client-install>`__.
```
Comment (rspurgeon) [inline kubernetes/docs/includes/deploy-jars-k8s.rst:11]: I like this but I also liked the text that discusses Converters and SMTs, as their JARs also need to be deployed even if it's not a full custom connector.   As it's the same exact problem/workflow, can we add back in some language about those features of Connect as well to try and help any readers that end up here looking to deploy a custom SMT w/ Operator?

File: kubernetes/docs/includes/deploy-jars-k8s.rst (hunk 1 above)
Comment (ybyzek) [reply kubernetes/docs/includes/deploy-jars-k8s.rst]: @rspurgeon IIUC, you are pointing out that this links to a docs page that doesn't have a link to CHub, right?  If so, please file a DOCS jira to get it fixed ;)

Comment (rspurgeon) [reply kubernetes/docs/includes/deploy-jars-k8s.rst]: Not exactly, I meant that by using the doc replacement for |c-hub| it just puts in the text, but there is no direct link from here to https://www.confluent.io/hub/

It's a nit, but might save someone a google search so seemed helfpul

File: kubernetes/docs/includes/deploy-jars-k8s.rst (hunk 2 above)
Comment (ybyzek) [reply kubernetes/docs/includes/deploy-jars-k8s.rst:11]: Please see latest commit, I added the links back in (at the top)

File: kubernetes/docs/includes/deploy-jars-k8s.rst (hunk 1 above)
Comment (ybyzek) [reply kubernetes/docs/includes/deploy-jars-k8s.rst]: Ack and fixed!

**Review by rspurgeon — APPROVED** [review]
LGTM, thank you

**Pull request merged**